wcivf/apps/elections/models.py

```diff
@@ -457,30 +457,39 @@ def expected_sopn_date(self):
             return get_election_timetable(
                 self.ballot_paper_id, self.post.territory
             ).sopn_publish_date
-
         except AttributeError:
             return None
 
     @property
     def registration_deadline(self):
-        date = get_election_timetable(
-            self.ballot_paper_id, self.post.territory
-        ).registration_deadline
+        try:
+            date = get_election_timetable(
+                self.ballot_paper_id, self.post.territory
+            ).registration_deadline
+        except AttributeError:
+            return None
```

Comment:
There's more info on this in the commit message for https://github.com/DemocracyClub/WhoCanIVoteFor/commit/ec92c400dd1c54169cdd294727a09fd6b0bdd330

I don't really like what I've done here. Basically the motivation is that once I started calling `ballot.registration_deadline` in `get_global_registration_card()` tests started failing all over the place because we don't handle the case where `get_election_timetable()` returns `None` and we don't really set up "realistic" test data. I actually suspect this condition doesn't really exist with "real" data. I don't think we're handling this right at all, but I decided to just stop pulling the thread at this layer of abstraction and hit everything with the same hammer we hit `expected_sopn_date` with.

Reply:
Yeah, I'd like to see some decent documented exceptions from the timetables library, but more generally we should be using better test data in this project so we don't need to do this. Agree that's for another day. Can you write an issue?

Reply:
The package itself actually throws some fairly specific exceptions

https://github.com/DemocracyClub/uk-election-timetables/blob/b631dcea3e3ff7590ddca41cf680c3a72949fa05/uk_election_timetables/election_ids.py#L20-L53

The issue is that in `get_election_timetable()` we just catch them all (and some other errors) and return a null value. I just decided I needed to stop pulling the thread here.

I'll write up a ticket about it :+1: